Services/AssessmentQuestion/examples/class.exTestPlayerGUI.php

```diff
@@ -129,68 +117,112 @@ public function submitSolution()
 	{
 		global $DIC; /* @var ILIAS\DI\Container $DIC */
 		
-		// this can also be $_REQUEST or any other future ilias post-request handler
-		$serverRequestObject; /* @var \Psr\Http\Message\ServerRequestInterface $serverRequestObject */
-		
-		$questionId = 0; // initialise with id of question that just submits
-		
 		/**
-		 * fetch possibly existing participant solution, an empty one is required otherwise
+		 * initialise with id of question to be shown
 		 */
 		
-		$participantSolution = $this->getParticipantSolution($questionId);
+		$questionUuid = $DIC->assessment()->consumer()->questionUuid('any-valid-question-uuid');
+		$revisionUuid = $DIC->assessment()->consumer()->questionUuid('any-valid-revision-uuid');
 		
 		/**
-		 * let the solution object instance harvest the submission post data
+		 * fetch possibly existing participant answer uuid,
+		 * when no participant answer exist yet,
+		 * generate a new user answer uuid
 		 */
 		
-		$participantSolution->initFromServerRequest($serverRequestObject);
+		$userAnswerUuid = $this->getParticipantAnswerUuid($questionUuid->getId());
+		
+		if( $userAnswerUuid )
+		{
+			$userAnswerUuid = $DIC->assessment()->consumer()->userAnswerUuid($userAnswerUuid);
+		}
+		else
+		{
+			$userAnswerUuid = $DIC->assessment()->consumer()->newUserAnswerUuid();
+		}
 		
 		/**
-		 * get results calculator to be used to retrieve calculated reached points
-		 * that can be stored in a test result storage managed by the test object
+		 * generate a user answer submit containing the post data
 		 */
+		$asqPlayService = $DIC->assessment()->service()->play(
+			$this->buildAsqPlayServiceSpec(), $DIC->assessment()->consumer()->questionUuid($questionUuid),$DIC->assessment()->consumer()->revisionUuid($questionUuid,$revisionUuid)
+		);
+		
+		$DIC->assessment()->service()->play()->CreateUserAnswer(
```

Comment:
What I don't get (yet) by reading the examples and the README file is the relation between `CreateUserAnswer` and `GetUserScore`.
`CreateUserAnswer` does not return or process anything, yet. What exactly is the purpose of this method? Does it store the raw particpant data? Or is the consumer itself responsible for storing the data based on a (maybe) future return value of `CreateUserAnswer`?
The next interesting command is `GetUserScore`, which returns an `UserAnswerScoringContract` instance which can be used for result calculcation and storage. This method seems to be independent of `CreateUserAnswer`, so I guess it will always get the score from the persistence layer, right?

Please provide a few more sentences according to result storage and calculation in the README file.

Reply:
> CreateUserAnswer does not return or process anything, yet. What exactly is the purpose of 
> this method?
In the past, _user answer_ was called _user solution_. But my staff and I simply can't use this term. While a solution is the correct answer for us. :-D

CreateUserAnswer saves the answer given by a user. This must be a new answer to the question. If a previously given answer is to be updated UpdateUserAnswer must be chosen.

GetUserScore returns the score for the given answer from the point of view of the question sercice. The consumer specific settings like _For Each Questions Negative Points are set to '0 Points'_ are not considered here. This is a matter for the consumer.

I will update the Readme. There are errors in the proposal also.

CreateUserAnswer(...):int -> CreateUserAnswer(..):void

Reply:
I have done the updates https://github.com/ILIAS-eLearning/ILIAS/blob/e73fe5ea8692febf11c27b5f9262dd1d4047d4f9/Services/AssessmentQuestion/docs/readme.md

Thx for your questions and your time!

Reply:
> In the past, _user answer_ was called _user solution_. But my staff and I simply can't use this term. While a solution is the correct answer for us. :-D

I totally agree.

> CreateUserAnswer saves the answer given by a user. This must be a new answer to the question. If a previously given answer is to be updated UpdateUserAnswer must be chosen.
GetUserScore returns the score for the given answer from the point of view of the question sercice. The consumer specific settings like For Each Questions Negative Points are set to '0 Points' are not considered here. This is a matter for the consumer.

Thx for clarification and for updating the README file.